packages/browser-wallet/src/background/confirmation.ts

```diff
@@ -7,53 +7,82 @@ import {
     PendingWalletCredential,
     WalletCredential,
     NetworkConfiguration,
+    ConfirmedIdentity,
+    RejectedIdentity,
 } from '@shared/storage/types';
+import { not } from '@shared/utils/function-helpers';
+import { identityMatch } from '@shared/utils/identity-helpers';
+import { sleep } from 'wallet-common-helpers';
 import { IdentityTokenContainer, IdentityProviderIdentityStatus } from 'wallet-common-helpers/lib/utils/identity/types';
 import { updateCredentials, updateIdentities } from './update';
 
 const isPendingCred = (cred: WalletCredential): cred is PendingWalletCredential =>
     cred.status === CreationStatus.Pending;
 const isPendingIdentity = (identity: Identity): identity is PendingIdentity =>
     identity.status === CreationStatus.Pending;
-const updateInterval = 10000;
+const UPDATE_INTERVAL = 10000;
 
-async function monitorCredentialStatus(
-    jsonRpcUrl: string,
-    { deploymentHash, ...info }: PendingWalletCredential,
-    genesisHash: string
-) {
+type ShouldLoop = boolean;
+
+/**
+ * Takes a loopFun, that runs each UPDATE_INTERVAL, for as long as the loopFun resolves to true
+ */
+const loop = async (loopFun: () => Promise<ShouldLoop>) => {
+    const run = async () => {
+        if (await loopFun()) {
+            await sleep(UPDATE_INTERVAL).then(run);
+        }
+    };
+    await run();
+};
+
+let monitoredCredentials: PendingWalletCredential[] = [];
+const isCredEqualTo = (c1: PendingWalletCredential) => (c2: PendingWalletCredential) => c1.credId === c2.credId;
+const isMonitoringCred = (id: PendingWalletCredential): boolean => monitoredCredentials.some(isCredEqualTo(id));
+
+async function monitorCredentialStatus(jsonRpcUrl: string, cred: PendingWalletCredential, genesisHash: string) {
     const client = new JsonRpcClient(new HttpProvider(jsonRpcUrl, fetch));
-    async function loop() {
+    if (isMonitoringCred(cred)) {
+        return;
+    }
+
+    monitoredCredentials.push(cred);
+
+    const { deploymentHash, ...info } = cred;
+
+    await loop(async () => {
         const network = await storedCurrentNetwork.get();
         // Stop if the network has changed
         if (!network || network.genesisHash !== genesisHash) {
-            return;
+            monitoredCredentials = [];
```

Comment:
```suggestion
```
I don't think this works properly, actually. If I change network, and then this loops for the final time, it resets your list here, removing those added for the other network, right?

If you just remove this, it will resolve and then remove itself from the list afterwards, right?

Reply:
Same for identities.

Reply:
It is a bit hard to actually test without stagenet though :thinking:

Reply:
haha, yes.. I had something else that would reset the lists when changing network  as part of the network change handler.. the current implementation doesn't make any sense, you're right

Reply:
but I still think the reset needs to happen, due to the possibility of collission between identities/credentials across networks?

Reply:
Then have your monitoredX index under the network hash.